packages/next-codemod/bin/upgrade.ts

```diff
@@ -312,44 +329,78 @@ export async function runUpgrade(
+    // understanding of the codemods, we run all of the applicable codemods.
+    if (shouldRunReactCodemods) {
+      // https://react.dev/blog/2024/04/25/react-19-upgrade-guide#run-all-react-19-codemods
+      execSync(
+        // `--no-interactive` skips the interactive prompt that asks for confirmation
+        // https://github.com/codemod-com/codemod/blob/c0cf00d13161a0ec0965b6cc6bc5d54076839cc8/apps/cli/src/flags.ts#L160
+        `${execCommand} codemod@latest react/19/migration-recipe --no-interactive`,
+        { stdio: 'inherit' }
+      )
+    }
 
-  // To reduce user-side burden of selecting which codemods to run as it needs additional
-  // understanding of the codemods, we run all of the applicable codemods.
-  if (shouldRunReactCodemods) {
-    // https://react.dev/blog/2024/04/25/react-19-upgrade-guide#run-all-react-19-codemods
-    execSync(
-      // `--no-interactive` skips the interactive prompt that asks for confirmation
-      // https://github.com/codemod-com/codemod/blob/c0cf00d13161a0ec0965b6cc6bc5d54076839cc8/apps/cli/src/flags.ts#L160
-      `${execCommand} codemod@latest react/19/migration-recipe --no-interactive`,
-      { stdio: 'inherit' }
-    )
+    if (shouldRunReactTypesCodemods) {
+      // https://react.dev/blog/2024/04/25/react-19-upgrade-guide#typescript-changes
+      // `--yes` skips prompts and applies all codemods automatically
+      // https://github.com/eps1lon/types-react-codemod/blob/8463103233d6b70aad3cd6bee1814001eae51b28/README.md?plain=1#L52
+      execSync(`${execCommand} types-react-codemod@latest --yes preset-19 .`, {
+        stdio: 'inherit',
+      })
+    }
   }
 
-  if (shouldRunReactTypesCodemods) {
-    // https://react.dev/blog/2024/04/25/react-19-upgrade-guide#typescript-changes
-    // `--yes` skips prompts and applies all codemods automatically
-    // https://github.com/eps1lon/types-react-codemod/blob/8463103233d6b70aad3cd6bee1814001eae51b28/README.md?plain=1#L52
-    execSync(`${execCommand} types-react-codemod@latest --yes preset-19 .`, {
-      stdio: 'inherit',
-    })
-  }
   console.log() // new line
   if (codemods.length > 0) {
     console.log(`${pc.green('✔')} Codemods have been applied successfully.`)
   }
   endMessage()
+
+  if (dry) {
+    console.log() // new line
+    console.log(pc.bold('Dry-run Output:'))
+    console.log() // new line
+    console.log(
+      JSON.stringify(
+        {
+          revision,
+          verbose,
+          installedNextVersion,
+          installedReactVersion,
+          targetNextVersion,
+          targetReactVersion,
+          packageManager,
+          usesAppDir,
+          usesPagesDir,
+          isPureAppRouter,
+          isMixedApp,
+          shouldStayOnReact18,
+          codemods,
+          shouldRunReactCodemods,
+          shouldRunReactTypesCodemods,
+          execCommand,
+          dependenciesToInstall,
+          devDependenciesToInstall,
+        },
+        null,
+        2
+      )
+    )
+  }
```

Comment:
This just inevitably gets outdated and is noisy for most use cases i.e. would probably actually slow you down since you have to parse the whole output. The result of the dry run is the same output you get by normally running it. Just without persisting any changes.